src/hist/plot.py

```diff
@@ -72,26 +73,29 @@ def _expr_to_lambda(expr: str) -> Callable:
         varnames.append(tokval)
     varnames = list(OrderedDict.fromkeys([name for name in varnames if name != "x"]))
     lambdastr = f"lambda x,{','.join(varnames)}: {expr}"
-    return eval(lambdastr)
+    return eval(lambdastr)  # type: ignore
 
 
 def _curve_fit_wrapper(
-    func: Callable,
-    xdata: ArrayLike,
-    ydata: ArrayLike,
-    yerr: ArrayLike,
+    func: Callable[..., Any],
+    xdata: np.ndarray,
+    ydata: np.ndarray,
+    yerr: np.ndarray,
     likelihood: bool = False,
-) -> Tuple[ArrayLike, ArrayLike]:
+) -> Tuple[Tuple[float, ...], ArrayLike]:
     """
     Wrapper around `scipy.optimize.curve_fit`. Initial parameters (`p0`)
     can be set in the function definition with defaults for kwargs
     (e.g., `func = lambda x,a=1.,b=2.: x+a+b`, will feed `p0 = [1.,2.]` to `curve_fit`)
     """
     from scipy.optimize import curve_fit, minimize
 
-    p0 = None
-    if func.__defaults__ and len(func.__defaults__) + 1 == func.__code__.co_argcount:
-        p0 = func.__defaults__
+    params = list(inspect.signature(func).parameters.values())
+    p0 = [
+        1 if arg.default is inspect.Parameter.empty else arg.default
+        for arg in params[1:]
+    ]
+
```

Comment:
I'm using inspect.signature, as it's a more public interface; `__defaults__` (and maybe `__code__`) are untyped, AFAICT (or maybe just not always present on Callables). This also supports partial defaults, while the other was all or nothing.